doc/feature/morphology11.md

```diff
@@ -0,0 +1,128 @@
+        SECTION_GLIA_PROCESS = 2,
+        SECTION_GLIA_ENDFOOT = 3,
+        // ...
+    };
+
+    MorphologyVersion
+    {
+        // between existing ones
+        MORPHOLOGY_VERSION_H5_1 = 1,
+        MORPHOLOGY_VERSION_H5_2 = 2,
+        MORPHOLOGY_VERSION_H5_1_1 = 3,
+        // ...
+    };
+
+    class Morphology
+        // Default parameter brion::MorphologyRepairStage for all functions to
+        // brion::MORPHOLOGY_UNDEFINED. Needed only for V2 and throws there if
+        // brion::MORPHOLOGY_UNDEFINED.
+
+### New
+
+    /** The cell family represented by brion::Morphology. */
+    CellFamily
+    {
+        FAMILY_NEURON = 0,
+        FAMILY_GLIA = 1
+    };
+
+    class Morphology
+    {
+        ...
+        /**
+         * @return the cell family of that morphology. FAMILY_GLIA only for
+         *         MORPHOLOGY_VERSION_H5_1_1 with a perimeter dataset,
+         *         otherwise FAMILY_NEURON.
+         */
+        CellFamily brion::getCellFamily() const;
+
+        /**
+         * @return perimeters of the cross sections for each point of the
+         *         morphology in micrometers. Valid only for cell family
+         *         FAMILY_GLIA.
+         */
+        floatsPtr readPerimeters() const;
+
+        /**
+         * Write perimeters of morphology. The cell family will be set to
+         * FAMILY_GLIA.
+         *
+         * @param perimeters perimeters of the cross sections for each point of
+         *                   the morphology in micrometers
+         */
+        void writePerimeters( const floats& perimeters );
+
+## File format
+
+* new root 'cell_family' with datatype H5::NATIVE_INT
+* new root dataset 'perimeters' with dimension 1 and datatype H5::NATIVE_FLOAT
+* new root attributes 'creator', 'software_version', 'creation_time' with
+  datatype H5::C_S1 (string)
+* new root attribute 'version' with datatype H5::STD_U32LE and dimensionality 2
+
+## Examples
+
+    brion::Morphology glia( "glia.h5", brion::MORPHOLOGY_VERSION_H5_1_1 );
+    glia.writePoints( points );
+    glia.writeSections( sections );
+    glia.writeSectionTypes( types );
+    glia.writePerimeters( perimeters );
+
+    const brion::Morphology gliaRead( "glia.h5" );
+    BOOST_CHECK_EQUAL( gliaRead.getCellFamily(), brion::FAMILY_GLIA );
+
+    brion::Morphology neuron( "neuron.h5", brion::MORPHOLOGY_VERSION_H5_1_1 );
+    neuron.writePoints( points );
+    neuron.writeSections( sections );
+    neuron.writeSectionTypes( types );
+
+    const brion::Morphology neuronRead( "neuron.h5" );
+    BOOST_CHECK_EQUAL( neuronRead.getCellFamily(), brion::FAMILY_NEURON );
+
+## Implementation
+
+TODO
+
+## Issues
+
+### 1: Do we choose cell family automatically or do we require an explicit selection?
+
+_Resolved: Yes_
+
+To maintain the existing API, the cell family will be chosen automatically. The
+family will be FAMILY_GLIA after writePerimeters() or the value of the
+cell_family enum, or FAMILY_NEURON in all other cases.
```

Comment:
You can add a new function and keep the semantics intact for old code. You just have to make FAMILY_NEURON the default. I prefer explicit rather than implicit in this case.